src/Microsoft.AspNet.OData.Shared/Query/HandleNullPropagationOptionHelper.cs

```diff
@@ -69,11 +69,11 @@ public static DataSourceProviderKind GetDataSourceProviderKind(this IQueryable q
             string queryProviderNamespace = query.Provider.GetType().Namespace;
             switch (queryProviderNamespace)
```

Comment:
@xuzhg Given that you added `DataSourceProviderKind.InMemory` should the default case still return `DataSourceProviderKind.None`

Reply:
Emm, maybe not. InMemory is not for the "IQueryable", this method is for the IQueryable. So, it's better to return "None". @gathogojr